warp10/src/main/java/io/warp10/script/functions/DEREF.java

```diff
@@ -76,13 +82,34 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
         } else if (i > 0 && statements.get(i) instanceof LOAD) {
           Object symbol = statements.get(i - 1);
           if (symbol instanceof String && values.containsKey(symbol)) {
-            statements.set(i - 1, NOOP);
-            statements.set(i, values.get(symbol));
+            Object value = values.get(symbol);
+
+            // Treat Macros in a specific way
+            if (value instanceof Macro) {
+              // If the macro contains a single element which is itself a macro, push
+              // that macro onto the stack as a shortcut
+              if (1 == ((Macro) value).size() && ((Macro) value).get(0) instanceof Macro) {
```

Comment:
Does it make sense to remove the `instanceof Macro` altogether and add any single statement from a Macro, whatever its type?

Reply:
Yep